src/language.ts

```diff
@@ -1,87 +1,1176 @@
+export type LanguageInfo = {
+  /**
+   * Two letter ISO code, such as `"en"` for English language.
+   * Can be `null` if the language is a flag to be evaluated at runtime,
+   * as it is the case for some "modes".
+   */
+  code: string | null;
+
+  /**
+   * The full OSM language flag, such as `"name:en"` for the English language.
+   * Can also be a non-OSM flag if the language needs to be evaluated at runtime, such as `"auto"`,
+   * as it is the case for some "modes".
+   */
+  flag: string;
+
+  /**
+   * English name of the language.
+   */
+  name: string;
+
+  /**
+   * Whether the language leverages only the latin charsets.
+   */
+  latin: boolean;
+
+  /**
+   * Some language descriptions corresponds to "modes" rather than to actual languages.
+   * For instance the "visitor" mode consists in displaying bylingual labels.
+   */
+  isMode: boolean;
+
+  /**
+   * Whether the language is compatible with the geocoding API
+   */
+  geocoding: boolean;
```

Comment:
What if there is a language supported only by Geocoding and not by map? Not sure if this is a case. Maybe there could be a set (bitmask?) of supported APIs: `apis: Api.MAP | Api.GEOCODING`. Just an idea.

Reply:
From what I could see in the Tilestat schema, all the geocoding languages are supported by Planet. It does not mean all the labels are available in all these languages, but at least some.

Reply:
I see. In the future if by some accident a support of a new language will be added to Geocoding which won't be supported by the map then we can introduce new property which will be a semantically minor change. I can't foresight another API that could be later supported by client-js and which will also use language, but I think that `apis` bitmaks would be then more appropriate, instead of multiple flags. That's my opinion :-)